src/snowflake/snowpark/mock/plan.py

```diff
@@ -167,6 +185,90 @@ def __init__(
         self.options = options
 
 
+def handle_order_by_clause(
+    order_by: List[Expression],
+    result_df: TableEmulator,
+    analyzer: "MockAnalyzer",
+    expr_to_alias,
+):
+    sort_columns_array = []
+    sort_orders_array = []
+    null_first_last_array = []
+    if order_by:
+        for exp in order_by:
+            sort_columns_array.append(analyzer.analyze(exp.child, expr_to_alias))
+            sort_orders_array.append(isinstance(exp.direction, Ascending))
+            null_first_last_array.append(
+                isinstance(exp.null_ordering, NullsFirst)
+                or exp.null_ordering == NullsFirst
+            )
+        kk = reversed(
+            list(zip(sort_columns_array, sort_orders_array, null_first_last_array))
+        )
+        for column, ascending, null_first in kk:
+            comparator = partial(custom_comparator, ascending, null_first)
+            result_df = result_df.sort_values(by=column, key=comparator)
+    return result_df
+
+
+def handle_range_frame_indexing(
+    window_spec,
+    res_index,
+    res,
+    analyzer,
+    expr_to_alias,
+    unbounded_preceding,
+    unbounded_following,
+):
+    if window_spec.order_spec:
```

Comment:
same, can we have a short desc on what this function is doing.
from the code looks like it's trying to find window accroding to the win spec, as it's returning a list of frames.